Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -641,6 +660,136 @@ namespace gdjs {
+        }
+
+        let highestY = this._findPlatformHighestYUnderObject(
+          platform,
+          upwardDeltaY,
+          downwardDeltaY
+        );
+        if (highestY < totalHighestY) {
+          totalHighestY = highestY;
+          highestGround = platform;
+        }
+      }
+      if (highestGround) {
+        const object = this.owner;
+        object.setY(object.getY() + totalHighestY - object.getAABB().max[1]);
+      }
+      return highestGround;
+    }
+
+    /**
+     * Find the highest Y of the floor reachable by the owner.
+     * @param platform The platform to be tested for collision.
+     * @param upwardDeltaY The owner won't move upward more than this value.
+     * @param downwardDeltaY The owner won't move downward more than this value.
+     */
+    private _findPlatformHighestYUnderObject(
+      platform: gdjs.PlatformRuntimeBehavior,
+      upwardDeltaY: float,
+      downwardDeltaY: float
+    ) {
+      const aabb = this.owner.getAABB();
+      const minX = aabb.min[0];
+      const maxX = aabb.max[0];
+      const minY = aabb.max[1] + upwardDeltaY;
+      const maxY = aabb.max[1] + downwardDeltaY;
+
+      const platformObject = platform.owner;
+      const platformAABB = platformObject.getAABB();
+      if (
+        platformAABB.max[0] <= minX ||
+        platformAABB.min[0] >= maxX ||
+        platformAABB.max[1] < minY ||
+        platformAABB.min[1] > maxY
+      ) {
+        return Number.MAX_VALUE;
+      }
+
+      let highestY = Number.MAX_VALUE;
+      for (const hitbox of platformObject.getHitBoxes()) {
+        let previousVertex = hitbox.vertices[hitbox.vertices.length - 1];
+        for (const vertex of hitbox.vertices) {
+          // Ignore edges that are too low
+          if (previousVertex[1] <= maxY || vertex[1] <= maxY) {
+            // Check vertex into the interval
+            if (minX <= vertex[0] && vertex[0] <= maxX) {
+              if (vertex[1] < minY && vertex[1] >= aabb.min[1]) {
+                // Platform is too high
+                return Number.MAX_VALUE;
+              }
+              // Ignore intersections that are too low
+              if (minY <= vertex[1] && vertex[1] <= maxY) {
+                highestY = Math.min(highestY, vertex[1]);
+              }
+            }
+            const deltaX = vertex[0] - previousVertex[0];
+            // Vertical edges doesn't matter
+            if (deltaX !== 0) {
+              // Check intersection on the left side of owner
+              if (
+                (vertex[0] < minX && minX < previousVertex[0]) ||
+                (previousVertex[0] < minX && minX < vertex[0])
+              ) {
+                const deltaY = vertex[1] - previousVertex[1];
+                const intersectionY =
+                  previousVertex[1] +
+                  ((minX - previousVertex[0]) * deltaY) / deltaX;
+                if (intersectionY < minY && intersectionY >= aabb.min[1]) {
+                  // Platform is too high
```

Comment:
Can you explain this if condition? I struggle to imagine what case it represents exactly.

Reply:
This mean "The floor is too high to follow but not over the object.". I added this because if the hitbox is in several parts the object could walk on one part and have another part over its head (I had the issue with the marching squares hitboxes, this exemple far from the right way of doing things but the case is still relevent).
I will add a comment and write the test.